update

```diff
@@ -33,7 +33,10 @@ fs.readdir(path.join(__dirname, '/bin'), function (err, files) {
   }
 
   function readBuiltinVersion (file) {
-    return solc(require(path.join(__dirname, '/bin', file))).version()
+    // NOTE: should be using this, but it leaks memory
+    // return solc(require(path.join(__dirname, '/bin', file))).version()
+    const filename = path.join(__dirname, '/bin', file)
+    return cp.execSync(`/usr/bin/env node -e "console.log(require('${filename}').cwrap('version', 'string', [])())"`).toString().trim()
```

Comment:
An added side-effect of this is it would fail if the binary can't be loaded or is missing a the version function (e.g. incompatible).

Reply:
This would be a problem because we have one executable (` soljson-v0.4.1-nightly.2016.9.9+commit.79867f49.js`) that crashes when you run it.

Reply:
Well, we could, but I'm still not sure if running the executables while generating the list is such a good idea. It's slow enough already and we still have this weird problem with nodejs not exiting immediately.

BTW, here's the error from that executable:
```
TypeError: Invalid Version: 0.4.1-nightly.2016.09.09+commit.79867f49.Emscripten.clang
    at new SemVer (/home/cameel/data/working-set/projects/sol/solc-js/node_modules/semver/semver.js:323:11)
    at compare (/home/cameel/data/working-set/projects/sol/solc-js/node_modules/semver/semver.js:614:10)
    at Function.gt (/home/cameel/data/working-set/projects/sol/solc-js/node_modules/semver/semver.js:643:10)
    at setupMethods (/home/cameel/data/working-set/projects/sol/solc-js/wrapper.js:20:27)
    at Object.<anonymous> (/home/cameel/data/working-set/projects/sol/solc-js/index.js:3:18)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
```
Looking at it now, I see that it's actually a problem in solc-js, not the executable so it should be fixable. The string it reports is probably just not valid semver. It seems to be the `09` in version. Later executables have no zeros and don't crash.

Reply:
https://github.com/ethereum/solc-js/issues/562